commentary/P3-commentary.txt

```diff
@@ -0,0 +1,12 @@
+                      Protocol update commentary
+
+Protocol Version: 3
+Builds on Protocol Version: 2
+
+This protocol update adds support for multiple addresses for the same
+account. In protocol version 3 each account has 16777216 addresses.
+
+This allows each account to have aliases for different uses and
+creates a kind of sub-account structure. An account owner can give out
+different aliases for different uses to keep track of transfers and
+assign them meaning.
```

Comment:
I think the commentary is a bit short. To add a bit more understanding, I've copied some snippets from other places below, as suggestions. 

- In protocol versions 1 and 2 accounts and account addresses are in 1-1
relationship. In protocol version 3 each account has 16777216 addresses, namely a so-called canonical account address together with matching account aliases. The canonical account address is derived when an account is created on chain. The other 16 million addresses with matching initial 29 bytes are referred to as account aliases for the same account. Thus, accounts can be referred to by any address whose initial 29 bytes match. 

- Each account still has one total account balance. Hence, transfers to and from aliases of an account add to and subtract from that total account balance, respectively. Transfers between different aliases of the same account do not change the balance of the account, apart from cost. Finalization, block, and baking rewards are always received on the account's canonical address.

- It is ensured that the protocol update is fully backwards compatible. 

- Because existing nodes do not support account aliases, node runners will have to update their nodes before the protocol update will take effect. The updated node software that supports the update will be released ahead of time to give node runners time to update. A new concordium-client tool will be released concurrently. Wallets do not require a new version to be compatible with the account alias functionality.

Reply:
I added this almost verbatim. Thanks.